query_functional_test.sh

```diff
@@ -21,29 +21,41 @@ DOCKER_IMAGE_TAG="us.gcr.io/model-159019/gh:$TAG"
 
 # Import data into graphhopper's internal format. It's necessary to move test data from /web before
 # running import, because import paths are relative to base /graphhopper folder, unlike in `mvn test`.
-# Similarly, move the car custom model yamls to ../, as that's how we've specified its location in the
-# configs to make mvn test happy
 docker run \
-    -v "$TMPDIR:/graphhopper/transit_data/"\
+    -v "$TMPDIR:/graphhopper/transit_data/graphhopper" \
     --rm \
      "$DOCKER_IMAGE_TAG" \
-     /bin/bash -c "cp -r ./web/test-data . && mv local_car_custom_model.yaml ../local_car_custom_model.yaml && \
-     mv freeway_car_custom_model.yaml ../freeway_car_custom_model.yaml && \
+     /bin/bash -c "cp -r ./web/test-data . && \
      java -Xmx2g -Xms1g -XX:+UseG1GC -XX:MetaspaceSize=100M \
-     -classpath web/target/graphhopper-web-1.0-SNAPSHOT.jar -server com.graphhopper.http.GraphHopperApplication import test_gh_config.yaml"
+     -classpath web/target/graphhopper-web-1.0-SNAPSHOT.jar -server com.graphhopper.http.GraphHopperApplication import ./transit_data/configs/test_gh_config.yaml"
 
 # Run link-mapping step
 docker run \
-    -v "$TMPDIR:/graphhopper/transit_data/"\
+    -v "$TMPDIR:/graphhopper/transit_data/graphhopper" \
+    -v "$TMPDIR:/graphhopper/transit_data/gtfs_link_mappings" \
     --rm \
     "$DOCKER_IMAGE_TAG" \
-    /bin/bash -c "mv local_car_custom_model.yaml ../local_car_custom_model.yaml && mv freeway_car_custom_model.yaml ../freeway_car_custom_model.yaml && \
+    /bin/bash -c "cp -r ./web/test-data . && \
     java -Xmx2g -Xms1g -XX:+UseG1GC -XX:MetaspaceSize=100M \
-    -classpath web/target/graphhopper-web-1.0-SNAPSHOT.jar com.graphhopper.http.GraphHopperApplication gtfs_links test_gh_config.yaml"
+    -classpath web/target/graphhopper-web-1.0-SNAPSHOT.jar com.graphhopper.http.GraphHopperApplication gtfs_links ./transit_data/configs/test_gh_config.yaml"
 
-# Run server in background
-docker run --rm --log-driver=none --name functional_test_server -p 50051:50051 -p 8998:8998 \
-    -v "$TMPDIR:/graphhopper/transit_data/" "$DOCKER_IMAGE_TAG" &
+# Run server in background (override standard CMD in Dockerfile.server with test_gh_config.yaml)
```

Comment:
wish we could override just the config file without having to repeat the entire command...maybe we could write the config path to an env variable or something, and the server could also accept the path from there? just an idea for a potential follow-up, definitely not a blocker

Reply:
That's a pretty cool idea! This whole `query_functional_test` script could use some cleanup in general, it's definitely messy to work with - having the config path live in an env var gives us a lot more flexibility re: how we organize things.

Keeping this as a potential follow-up but definitely will ponder!